### DISCUSSION TITLE
OTP Codes not being received by US Numbers

### DISCUSSION THREAD

**Discussion by metasoul111** [Web3Auth/Web3Auth#1298] [Help]
We have tried several phone numbers when testing OPT verification. Numbers from middle east and India are working fine. But, numbers from the US are not, in testing we are not getting OPT codes to US numbers. Here is the console logs of one of our testers who tried to use a US number.

[2023-02-08T15:09:56.755Z] INFO: {"browser":{"name":"Chrome","version":"109.0.0.0"},"os":{"name":"Linux"},"platform":{"type":"desktop"},"engine":{"name":"Blink"}} current browser info
fetch.ts:36          POST https://o503538.ingest.sentry.io/api/6524334/envelope/?sentry_key=7d794c60725a43418fe5dc89c4fb8f66&sentry_version=7&sentry_client=sentry.javascript.react%2F7.31.1 net::ERR_BLOCKED_BY_CLIENT
n @ fetch.ts:36
(anonymous) @ base.ts:77
add @ promisebuffer.ts:49
f @ base.ts:93
_sendEnvelope @ baseclient.ts:596
sendEvent @ baseclient.ts:319
sendEvent @ client.ts:103
(anonymous) @ baseclient.ts:554
(anonymous) @ syncpromise.ts:79
(anonymous) @ syncpromise.ts:183
(anonymous) @ syncpromise.ts:176
(anonymous) @ syncpromise.ts:97
s @ syncpromise.ts:58
then @ syncpromise.ts:69
_processEvent @ baseclient.ts:522
_captureEvent @ baseclient.ts:448
captureEvent @ baseclient.ts:184
(anonymous) @ hub.ts:241
_withClient @ hub.ts:474
captureEvent @ hub.ts:240
finish @ transaction.ts:219
finish @ idletransaction.ts:162
(anonymous) @ idletransaction.ts:220
i @ helpers.ts:98
setTimeout (async)
(anonymous) @ trycatch.ts:119
_startIdleTimeout @ idletransaction.ts:218
_popActivity @ idletransaction.ts:252
n @ idletransaction.ts:191
(anonymous) @ idletransaction.ts:37
(anonymous) @ request.ts:182
(anonymous) @ request.ts:136
p @ instrument.ts:96
(anonymous) @ instrument.ts:155

**Comment by ryanberckmans** [comment]
I can confirm that we're seeing this, too. OTP codes sent to a US number in testnet or mainnet are not received by the phone.

We aren't seeing any errors in the javascript console. Instead, Web3Auth says, as expected, that the OTP code has been sent to the phone, and asks for the OTP code to be typed in, but no code is ever received by the phone.

This seems like a possible SMS routing/blacklist issue and not related to the web side, but that's a guess.

EDIT: we may not be receiving SMS codes to any numbers. We are using `Web3AuthCore`, `Web3AuthConnector`, `loginProvider: 'sms_passwordless'`, and tested both US and international numbers using both Web3Auth Plug N'Play testnet and cyan mainnet. Here's the site we used to test international numbers https://receive-smss.com/sms/2348153353131/

**Comment by chaitanyapotti** [comment]
We're still awaiting approvals from US govt. reg. toll free number activation. SMS login by us is still not in prod.
Alternatively, you can bring your own sms login provider (via twilio/auth0, cognito, firebase etc) and work with web3auth

**Reply by ryanberckmans** [reply]
Great to know, thanks. Looking forward to this becoming available.
